src/audiovideocontroller/AudioVideoControllerState.ts

```diff
@@ -162,4 +171,63 @@ export default class AudioVideoControllerState {
   maxVideoTileCount: number = 0;
 
   startTimeMs: number | null = null;
+
+  /*
+   * Reset state corresponding to state that is dependent on a individual connection
+   * and may not be valid for others, e.g. on a reconnection.
+   */
+  resetConnectionSpecificState(): void {
+    // For auditing reasons, we will comment on the state that we do not touch here. Note that `DefaultAudioVideoController.actionConnect`
+    // also resets certain state, some to cached members:
+    // Reset to empty/null/new state: `browserBehavior`, `transceiverController`, `volumeIndicatorAdapter`, `enableSimulcast`
+    //                                `signalingOpenDurationMs`, `iceGatheringDurationMs`, `startAudioVideoTimestamp`, `attendeePresenceDurationMs`
+    //                                `meetingStartDurationMs`, `startTimeMs`, `lastKnownVideoAvailability`, `videoCaptureAndEncodeParameter`, `videosToReceive`
+    //                                `videosPaused`, `videoStreamIndex`, `statsCollector`, `connectionMonitor`
+    // Reset to existing/cached values: `logger`, `meetingSessionConfiguration`, `realtimeController`, `mediaStreamBroker`,
+    //                                  `audioMixController`, `reconnectController, `audioProfile`, `eventController`
+
+    // `signalingClient` can be reused from a failed/disconnected state.
+
+    if (this.peer) {
+      this.peer.close();
+    }
+    this.peer = null;
+    this.previousSdpOffer = null;
+    this.sdpOfferInit = null;
+
+    // `audioVideoController` members should either be reusable, or moved to `AudioVideoControllerState` and
+    // cleaned up here.
+
+    // We don't want to mutate `videoTileController` as most video tiles will still be there on reconnect. We can remove tiles on the
+    // first index we receive if they no longer exist
+
+    // `mediaStreamBroker`, `activeAudioInput`, and `activeVideoInput` are cleaned up seperately in `ReleaseMediaStreamsTask`
```

Comment:
Didn't `ReleaseMediaStreamTask` no longer exists?

Reply:
Hah, you are right, I must have been looking at old branches. Let me fix up this comment